build_runner/lib/src/generate/create_merged_dir.dart

```diff
@@ -127,13 +115,21 @@ Future<bool> _createMergedOutputDir(
     var packagesFileContent = packageGraph.allPackages.keys
         .map((p) => '$p:packages/$p/')
         .join('\r\n');
-    for (var dir in rootDirs) {
-      var packagesAsset = new AssetId(root, p.url.join(dir, '.packages'));
+
+    if (root != null) {
+      var packagesAsset = new AssetId(root, '.packages');
```

Comment:
I think you want `packageGraph.root.name` instead of `dir`.

Reply:
Done. Added an assert.